### PR TITLE
feat: Refactor selfDestruct integration tests

### LINKED ISSUE #8379: Refactor `SelfDestructOperationTest`
As the description of the WP suggests, refactor the mentioned integration tests in web3 module (`SelfDestructOperationTest`, `ContractCallDynamicCallsTest` and `ContractCallEvmCodesTest`) in a way to be self-sufficient enough and rely only on the provided data in this specific test. Essentially, if some of the data in separate tests inside the same integration test file is repeated, it could abstracted away in the integration test class, if appropriate

### PR DESCRIPTION
**Pull request by kselveliev**
**Description**:
This PR refactor test in the SelfDestructOperationTest to use the new web3j plugin.

This PR modifies:
SelfDestructOperationTest is now refactored 
Deleted bin,sol,json files that were used by the old approach
Added a new deployer in the testWeb3jService which allows deploying contracts that have a payable constructor.

**Related issue(s)**:

Fixes #8379 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/operations/SelfDestructOperationTest.java [hunk 1]
```diff
@@ -62,62 +46,19 @@ void testSuccessfulExecute() {
...
-
-        domainBuilder
-                .recordFile()
-                .customize(f -> f.bytes(selfDestructContractBytes))
-                .persist();
-
-        return selfDestructContractEntity.toEntityId();
-    }
+        final var contract = testWeb3jService.deployWithValue(SelfDestructContract::deploy, BigInteger.valueOf(1000));
+        final var functionCall = contract.send_destructContract(systemAccountAddress.toUnprefixedHexString());
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/operations/SelfDestructOperationTest.java:59]: Can't we use `call_destructContract`?

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/evm/contracts/operations/SelfDestructOperationTest.java:59]: Yeah that's a good question. Seems like even with the generateBoth flag the task only generates send_destructContract and there is no call method here. I haven't really researched why this happens.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**